File: python/atelier_bench/metrics/labor_cost.py
"""labor-cost metric."""

from __future__ import annotations

from typing import TYPE_CHECKING

from ..contracts.metric_contract import MetricUnit
from ..support.abstract_metric import AbstractMetric

if TYPE_CHECKING:
    from ..dataset import Dataset


class LaborCostMetric(AbstractMetric):
    KEY = "labor-cost"

    def __init__(self) -> None:
        super().__init__(self.KEY, MetricUnit.CENTS)

    def compute(self, data: "Dataset") -> float:
        return data.labour_minutes() * 125

File: go/atelier/metrics/labor_cost.go
// Package metrics carries the 16 metrics. Each embeds support.BaseMetric,
// declares only Compute, and registers itself in an init function.
package metrics

import (
	"atelier.example/lane/atelier/contracts"
	"atelier.example/lane/atelier/dataset"
	"atelier.example/lane/atelier/support"
)

// Compile-time proof of satisfaction; nothing else names the contract here.
var _ contracts.MetricContract = (*LaborCostMetric)(nil)

// LaborCostMetric computes the labor-cost metric.
type LaborCostMetric struct {
	support.BaseMetric
}

// NewLaborCostMetric builds the metric with its key and unit.
func NewLaborCostMetric() *LaborCostMetric {
	return &LaborCostMetric{BaseMetric: support.NewBaseMetric("labor-cost", contracts.UnitCents)}
}

// Compute is the only contract method this type declares; Key and Unit are
// promoted from the embedded base.
func (m *LaborCostMetric) Compute(data *dataset.Dataset) float64 {
	return float64(data.LabourMinutes() * 125)
}

// init registers this metric.
func init() {
	Register(NewLaborCostMetric())
}
